### ISSUE TITLE
Enable group chats in release build

### ISSUE THREAD

**Issue by lukaszfryc** [status-im/status-react#6882]
Enable group chats and pairing in release builds and cherry pick it into release/0.9.32 branch. Both features should remain under development mode.

Please update release changelog afterwards https://notes.status.im/mobile-release-notes?both

**Comment by lukaszfryc** [comment]
@cammellos could you share your thoughts on what's needed to release the group chats? Is the core work already done?

**Comment by cammellos** [comment]
@lukaszfryc if no blocker have been reported, I would just want to push out a couple of commits to fortify the implementation but otherwise we are good to go.
I would release them in development mode at first & then after a couple of weeks of feedback enable them.

What do you reckon?

**Comment by chadyj** [comment]
@cammellos Whilst on the topic can you outline the PFS release plan too?

**Comment by cammellos** [comment]
So, if no issues have been reported on group chats & pairing, I think we can start thinking about releasing PFS and getting things ready.
1) Pairing as it is released is already fully pfs, so no work to be done.

2) 1-to-1 we need to handle a few cases https://discuss.status.im/t/pfs-and-missing-contact-information/362/24 & https://discuss.status.im/t/pfs-and-wallet-recovery/335 which will require some UI work.

3) Group chats to be fully PFS will also require some ux changes (probably in the form of having  to accept the chat invitation), but this will be done after 2 as  2 is easier to handle.

Enabling PFS for 1-to-1 will be a breaking change, but if we release group chats (or pairing, or both ) next week, those clients will be already compatible, so it will give us a bit more time for users to upgrade, making the upgrade less painful.

<blockquote><img src="https://discuss.status.im/letter_avatar_proxy/v2/letter/p/b2d939/200.png" width="48" align="right"><div><img src="/uploads/default/original/1X/8c298ce8e38884365c17cdf0afea1ae108e992f9.png" height="14"> Status.im</div><div><strong><a href="https://discuss.status.im/t/pfs-and-missing-contact-information/362/24">PFS and missing contact information</a></strong></div><div>Thanks again @cammellos for going through all the explanations. So it sounds like we will always have this use case for when Alice tries to contact Bob using a public key? Sorry if I misunderstood this point.  I’m all for the approach and thanks as well to @denis and @maciej for putting together designs so quickly. To follow up on Hester’s question:   what does Chad see as long as the contact request has not been accepted?   Do we allow Chad to continue the conversation with a clear icon/visual ...</div></blockquote>
<blockquote><img src="https://discuss.status.im/uploads/default/original/1X/8c298ce8e38884365c17cdf0afea1ae108e992f9.png" width="48" align="right"><div><img src="/uploads/default/original/1X/8c298ce8e38884365c17cdf0afea1ae108e992f9.png" height="14"> Status.im</div><div><strong><a href="https://discuss.status.im/t/pfs-and-wallet-recovery/335">PFS and wallet recovery</a></strong></div><div>Hello everyone,  as you know we are working on PFS. The main idea is that we will exchange an x3dh bundle and encrypt message using the double ratchet algorithm.  https://signal.org/docs/specifications/x3dh/  https://signal.org/docs/specifications/doubleratchet/  A problem that we need to solve is wallet recovery and how to handle this.  When a user recover their wallet they will lose all the contacts and messages.  Currently if a contact sends a message to a user who has just recovered their ac...</div></blockquote>

**Comment by lukaszfryc** [comment]
>  if no blocker have been reported, I would just want to push out a couple of commits to fortify the implementation but otherwise we are good to go.

@cammellos there are no blockers. Only [issues](https://github.com/status-im/status-react/labels/group-chat) we already discussed that can be fixed later.

Re releasing group chats in dev mode, I don't think anyone will use them this way. So, we will not learn anything besides just having the code enabled in the release build. I'd suggest to enable them in normal mode for nightlies and include in the next release. @cammellos?

**Comment by cammellos** [comment]
@lukaszfryc I am ok with that, the only reason to have them in dev mode would be to warn users that is a beta-feature. 

So as long as we warn users that is a beta feature, I am ok in having them always on, in order to manage expectations.

**Comment by cammellos** [comment]
What about pairing, should we do the same? release it I mean? The only thing I would want to add is a way for the user to name its device, but other than that if no issues/blockers I'd be happy to release

**Comment by chadyj** [comment]
+1 for rolling this out. We can be sure to communicate to users this is an early feature and testing and feedback is appreciated.

**Comment by lukaszfryc** [comment]
If we still consider it a beta-feature, it makes sense to put it under dev mode. I think it's the best way to indicate (inside the app) that something is a beta feature.

**Comment by cammellos** [comment]
@lukaszfryc what about something like:
`Start group chat (Beta)`? We can also add a `Beta` tag somewhere in the actual group chat, it feels that way it's clear

**Comment by chadyj** [comment]
Note, the whole app is in beta! Every feature is a beta :)

But it won't hurt doing `Start group chat (Beta)`. :-)

**Comment by lukaszfryc** [comment]
> `Start group chat (Beta)`? We can also add a `Beta` tag somewhere in the actual group chat, it feels that way it's clear

Hm, I first thought that "dev mode" is for beta-features. But, we can actually differentiate between "options for developers" (dev mode) and beta features. 

> Note, the whole app is in beta! Every feature is a beta :)

Yeah. I'd say that if there is no major issue that we want to warn about, I'd not bother users with "Beta" word in yet another place.

**Comment by lukaszfryc** [comment]
> What about pairing, should we do the same? release it I mean? The only thing I would want to add is a way for the user to name its device, but other than that if no issues/blockers I'd be happy to release

@churik I see we have some "pairing" issues that are leftovers after https://github.com/status-im/status-react/pull/6697. Is any of them a blocker for releasing the pairing in current state?

**Comment by cammellos** [comment]
ok, in any cases, we are all in agreement about releasing next week, when are we cutting the release branch these days? sometimes early next week? just to get things organized

**Comment by lukaszfryc** [comment]
> ok, in any cases, we are all in agreement about releasing next week, when are we cutting the release branch these days? sometimes early next week? just to get things organized

@cammellos let's discuss it in #6891

**Comment by churik** [comment]
@lukaszfryc 

> @churik I see we have some "pairing" issues that are leftovers after #6697. Is any of them a blocker for releasing the pairing in current state?

I don't think so if both Group chats and syncing were under development mode.

**Comment by lukaszfryc** [comment]
@churik so you mean we can release pairing? Btw, for group chats, we are planning to use "Start group chat (Beta)" to indicate that it is a beta-feature. Do you think we should do the same for "Devices (Beta)"?

**Comment by lukaszfryc** [comment]
According to https://github.com/status-im/status-react/issues/6891, the release branch will be created today. @cammellos could you go ahead and enable group chats in release builds?

As for pairing, I still think we should improve on UX there (see https://github.com/status-im/status-react/issues/6898#issuecomment-442116620). At least [update copies](https://github.com/status-im/status-react/issues/6898#issuecomment-442061769). But if we do not have time, we may include it in the release and see if someone will use it.

**Comment by churik** [comment]
@lukaszfryc I think it should be under development mode, as this feature is still WIP.

**Comment by lukaszfryc** [comment]
@cammellos I updated description of this issue and assigned it to you.

**Comment by lukaszfryc** [comment]
group chats and pairing were enabled in https://github.com/status-im/status-react/commit/dd21f6a4725211cf76e666015bea760a844cf315

**Comment by chadyj** [comment]
Awesome!
Release notes updated.
